lib/replace.mjs

```diff
@@ -27,10 +28,16 @@ export default function create_rule (md, emojies, shortcuts, scanRE, replaceRE)
         if (offset + match.length < src.length && !ZPCc.test(src[offset + match.length])) {
           return
         }
-      } else {
+      } else if (match[0] === ':' && match[match.length-1] === ':') {
+        // emoji_name specified like :smile:
         emoji_name = match.slice(1, -1)
+      } else {
+        // replace unicode emoji using canonical name
+        emoji_name = emojies_rev[match]
       }
 
+      if (!emoji_name) { return }
```

Comment:
Not strictly necessary, but in case the regex matches something for which an `emoji_name` can't be found, just don't replace anything.